integration/spark/src/main/scala/org/apache/spark/sql/execution/strategy/CarbonLateDecodeStrategy.scala

```diff
@@ -349,8 +349,15 @@ private[sql] class CarbonLateDecodeStrategy extends SparkStrategy {
       scanBuilder: (Seq[Attribute], Seq[Expression], Seq[Filter], Seq[PartitionSpec])
         => RDD[InternalRow]): CodegenSupport = {
     val table = relation.relation.asInstanceOf[CarbonDatasourceHadoopRelation]
-    val extraRdd = MixedFormatHandler.extraRDD(relation, rawProjects, filterPredicates,
-      new TableStatusReadCommittedScope(table.identifier, FileFactory.getConfiguration),
+    val isTransactionalTable = table.carbonTable.isTransactionalTable
+    val readCommittedScope = new TableStatusReadCommittedScope(
```

Comment:
This is only used to set in CarbonScanRDD, right?
Can we move it to the constructor of CarbonScanRDD? Then we do not need to set it explicitly

Reply:
I thought about it, but multiple method signature need to modified to pass readcommitted scope till scanRDD.
So, I used set method. 

Currrently, directFill and vectorReader variables are also set similar way

Reply:
You mean many places created the readcommitted scope by themselves? But in that case, it will be hard to avoid using wrong committed scope in scanRDD.getPartition, like maybe in some flow someone created a scanRDD but forget to set the read committed scope.

Reply:
In carbonScanRDD line 140, I have added check for tableInputFormat instance. so, it won't use wrong readcommitted scope.
Also only TableInputFormat case will use TableStatusReadCommittedScope, other case it is LatestFileReadCommittedScope.